chatsky/core/vector_store.py

```diff
@@ -0,0 +1,188 @@
+from typing import Optional, List, Dict, Any
+from pydantic import BaseModel
+import logging
+from uuid import uuid4
+from context import Context
+from message import Message
+from service import Service
+from langchain.vectorstores import Chroma
+from langchain.embeddings import HuggingFaceEmbeddings
+from langchain.schema import Document
+
+logger = logging.getLogger(__name__)
+
+
+class VectorStoreConfig(BaseModel):
+    """Configuration model for vector store initialization."""
+
+    collection_name: str = "default"
+    """Name of the collection in vector database. Defaults to "default"."""
+    top_k: int = 3
+    """Number of top results to return during retrieval. Defaults to 3."""
+    persist_directory: Optional[str] = None
+    """Directory path for persistent storage. Defaults to None."""
+    model_name: str = "DeepPavlov/rubert-base-cased"
+    """Name of the embedding model. Defaults to "DeepPavlov/rubert-base-cased"."""
+
+
+class VectorStoreService:
```

Comment:
Inheriting this class from `abc.ABC` and decorating its abstract methods with `abc.abstractmethod` would forbid instantiating this class or any of its subclasses with abstract methods undefined.

This would ensure that any instance of this class has these methods defined.

There would also be no need to mention `:raises NotImplementedError: Must be implemented by subclasses.` in abstract method docstrings since the decorator would indicate that they are abstract.